crates/engine/tree/src/tree.rs

```diff
@@ -359,6 +359,27 @@ pub enum TreeAction {
     MakeCanonical(B256),
 }
 
+/// Represents the state of the backfill synchronization process.
+#[derive(Debug, PartialEq, Eq, Default)]
+enum BackfillSyncState {
```

Comment:
cool, I can see this being useful for more handler impls, let's move this to backfill.rs?

Reply:
Yeah, just moved